### DISCUSSION TITLE
ReportLeakedContainers shoud ignore testcontainers/sshd

### DISCUSSION THREAD

**Discussion by durban77** [trinodb/trino#21969] [General]
When you expose a port to the network in testing with 
`org.testcontainers.Testcontainers#exposeHostPorts(int...)`
It starts a `testcontainers/sshd` in the docker session as can be seen in `org.testcontainers.containers.PortForwardingContainer`.
After the tests complete, the new `testing/trino-testing-containers/src/main/java/io/trino/testing/containers/junit/ReportLeakedContainers.java`'s check will crash the JVM. 
I think that ReportLeakedContainers should ignore the sshd, as it is out of scope for this check.

**Comment by findepi** [comment]
Which test is affected?

**Comment by durban77** [comment]
My tests, not Trino's own tests.

`java.util.ServiceLoader` loads `org.junit.platform.launcher.TestExecutionListener` services, and grabs `io.trino.testing.containers.junit.ReportLeakedContainers$Listener` as specified in `/META-INF/services/org.junit.platform.launcher.TestExecutionListener`. 
And it is in my classpath, because I'm using the fine codes from trino-testing-containers (`io.trino.testing.containers.TestContainers`).

Therefore, it will check my containers also, and I'm happy about it, just there's a scenario where exposed host port to the docker network is implemented with an sshd (testcontainers/sshd:1.1.0) internally, and that triggers the JVM kill (`io.trino.testing.services.junit.Listeners#reportListenerFailure`).

I haven't found a nicer approach yet to handle this situation, so I'm using `System.setProperty("ReportLeakedContainers.disabled", "true"); @AfterAll`

**Comment by findepi** [comment]
since you use trino-testing-containers, including `ReportLeakedContainers.Listener`, would it work to call `ReportLeakedContainers.ignoreContainerId` on the sshd container?

**Reply by durban77** [reply]
If I could just clean up after `exposeHostPorts` call, that would be a great.
`org.testcontainers.containers.PortForwardingContainer.INSTANCE` is not telling me the containerId to do so.

**Comment by jja725** [comment]
I'm also having the similar issue here, any update to this? @durban77 @findepi 
```
2024-08-12T16:27:25.450-0600	INFO	main	io.trino.testing.containers.junit.ReportLeakedContainers	Checking for leaked containers
2024-08-12T16:27:25.466-0600	INFO	main	stderr	FATAL: io.trino.testing.containers.junit.ReportLeakedContainers$Listener: Leaked containers: [container{id=f0abe085dd48fc3e63bea434d6aca42c3986d8cfe494b82c3c23e009de18db58, image=testcontainers/sshd:1.2.0, imageId=sha256:3bc9c981918e8ddec0b0f9454cd604bd2e7b1775602894846181b275580b7401}]
2024-08-12T16:27:25.466-0600	INFO	main	stderr	JVM will be terminated
```

**Comment by findepi** [comment]
addressed in https://github.com/trinodb/trino/pull/26946

**Reply by durban77** [reply]
Many thanks! Just checked and it works fine on my codebase!
